cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/StorageManager.java

```diff
@@ -93,4 +93,28 @@ public Storage getStorage(StorageType.Type storageType) {
     throw new IllegalArgumentException(
         "No configured storage found for type: " + storageType.getValue());
   }
+
+  /**
+   * Get the storage from the provided path. Iterate over all storages and return the storage which
+   * is configured and its endpoint is the prefix of supplied path. Note: if local storage type is
+   * configured, then return the local storage. This is a workaround for unit tests to work because
+   * hdfs and local storage do not append scheme to the paths. See:
+   * https://github.com/linkedin/openhouse/issues/121
+   *
+   * @param path Path that contains the scheme
+   * @return the storage
+   */
+  public Storage getStorageFromPath(String path) {
+    for (Storage storage : storages) {
+      if (storage.isConfigured()) {
+        if (StorageType.LOCAL.equals(storage.getType())) {
```

Comment:
Shouldn't this be fallback only if path does not start with any other configured endpoint?
What if the path is an S3 storage path but local storage is also configured for some other tables?

Reply:
done. Please check